### PR TITLE
test: Updates to e2e tests. Removing AzDriverNode creation on setup/teardown and running scheduler extender tests only for v2 driver

### PR DESCRIPTION
**Pull request by landreasyan**
…rnode create/delete before/after controller/scheduler tests.

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by landreasyan** [comment]
Pushing these changes for visibility of the bug addressed in this PR. Controller and scheduler tests were deleting AzDriverNode instances which was causing tests running afterwards to fail. Similarly, scheduler extender was being added to v1 driver tests as well, causing another failure. 

I've been running all testing configurations since yesterday to verify that there are no further issues. V1 tests are ok but I am now seeing some failures in v2 tests that I want to debug and address in this PR.

File: test/e2e/controller_test.go [hunk 1]
```diff
@@ -143,6 +129,8 @@ var _ = ginkgo.Describe("Controller", func() {
 
 			err = azDiskClient.DiskV1alpha1().AzDriverNodes(namespace).Delete(context.Background(), nodes[0], metav1.DeleteOptions{})
 			framework.ExpectNoError(err)
+			// re-create the node after test execution
+			defer testsuites.NewTestAzDriverNode(azDiskClient.DiskV1alpha1().AzDriverNodes(namespace), nodes[0])
```
Comment (sunpa93) [inline test/e2e/controller_test.go:133]: this change is already in #31

File: test/e2e/controller_test.go [hunk 2]
```diff
@@ -187,6 +175,8 @@ var _ = ginkgo.Describe("Controller", func() {
 			// fail primary attachment
 			err = azDiskClient.DiskV1alpha1().AzDriverNodes(namespace).Delete(context.Background(), primaryNode, metav1.DeleteOptions{})
 			framework.ExpectNoError(err)
+			// re-create the node after test execution
+			defer testsuites.NewTestAzDriverNode(azDiskClient.DiskV1alpha1().AzDriverNodes(namespace), nodes[0])
```
Comment (sunpa93) [inline test/e2e/controller_test.go:179]: same goes for this line

**Review by sunpa93 — CHANGES_REQUESTED** [review]

File: test/e2e/azdiskschedulerextender_test.go [hunk 3]
```diff
@@ -40,9 +40,10 @@ var _ = ginkgo.Describe("AzDiskSchedulerExtender", func() {
 		schedulerExtenderTests(false)
 	})
 
-	ginkgo.Context("[multi-az]", func() {
-		schedulerExtenderTests(true)
-	})
+	// TODO add support for multi-az
```
Comment (edreed) [inline test/e2e/azdiskschedulerextender_test.go:43]: Be sure the create an issue for this.

**Review by edreed — APPROVED** [review]

**Review by sunpa93 — APPROVED** [review]

File: test/e2e/azdiskschedulerextender_test.go (hunk 3 above)
Comment (landreasyan) [reply test/e2e/azdiskschedulerextender_test.go:43]: I just created it: https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/issues/51

**Pull request merged**